src/core/server/elasticsearch/types.ts

```diff
@@ -28,58 +28,46 @@ import { ServiceStatus } from '../status';
-   *
-   * @param type Unique identifier of the client
-   * @param clientConfig A config consists of Elasticsearch JS client options and
-   * valid sub-set of Elasticsearch service config.
-   * We fill all the missing properties in the `clientConfig` using the default
-   * Elasticsearch config so that we don't depend on default values set and
-   * controlled by underlying Elasticsearch JS client.
-   * We don't run validation against the passed config and expect it to be valid.
-   *
-   * @example
-   * ```js
-   * const client = elasticsearch.createCluster('my-app-name', config);
-   * const data = await client.callAsInternalUser();
-   * ```
-   */
-  readonly createClient: (
-    type: string,
-    clientConfig?: Partial<ElasticsearchClientConfig>
-  ) => ICustomClusterClient;
-
-  /**
-   * @deprecated
-   * Use {@link ElasticsearchServiceStart.legacy | ElasticsearchServiceStart.legacy.client} instead.
-   *
-   * A client for the `admin` cluster. All Elasticsearch config value changes are processed under the hood.
-   * See {@link IClusterClient}.
-   *
-   * @example
-   * ```js
-   * const client = core.elasticsearch.adminClient;
-   * ```
-   */
-  readonly adminClient: IClusterClient;
+  legacy: {
+    /**
+     * @deprecated
+     * Use {@link ElasticsearchServiceStart.legacy | ElasticsearchServiceStart.legacy.createClient} instead.
+     *
+     * Create application specific Elasticsearch cluster API client with customized config. See {@link IClusterClient}.
+     *
+     * @param type Unique identifier of the client
+     * @param clientConfig A config consists of Elasticsearch JS client options and
+     * valid sub-set of Elasticsearch service config.
+     * We fill all the missing properties in the `clientConfig` using the default
+     * Elasticsearch config so that we don't depend on default values set and
+     * controlled by underlying Elasticsearch JS client.
+     * We don't run validation against the passed config and expect it to be valid.
+     *
+     * @example
+     * ```js
+     * const client = elasticsearch.createCluster('my-app-name', config);
+     * const data = await client.callAsInternalUser();
+     * ```
+     */
+    readonly createClient: (
+      type: string,
+      clientConfig?: Partial<ElasticsearchClientConfig>
+    ) => ICustomClusterClient;
 
-  /**
-   * @deprecated
-   * Use {@link ElasticsearchServiceStart.legacy | ElasticsearchServiceStart.legacy.client} instead.
-   *
-   * A client for the `data` cluster. All Elasticsearch config value changes are processed under the hood.
-   * See {@link IClusterClient}.
-   *
-   * @example
-   * ```js
-   * const client = core.elasticsearch.dataClient;
-   * ```
-   */
-  readonly dataClient: IClusterClient;
+    /**
+     * @deprecated
+     * Use {@link ElasticsearchServiceStart.legacy | ElasticsearchServiceStart.legacy.client} instead.
```

Comment:
nit: I don't think this line applies anymore?

Reply:
We can remove the comment when the new elasticsearch client merged. Till that moment, we'd prefer plugins to use start contract. Wouldn't we?

Reply:
Ah I misread this. I thought it was copied over from dataClient. 👍